sw/host/opentitanlib/src/io/spi.rs

```diff
@@ -150,7 +150,9 @@ pub trait Target {
     fn get_max_transfer_count(&self) -> Result<usize>;
 
     /// Maximum chunksize handled by this SPI device.
-    fn max_chunk_size(&self) -> Result<usize>;
+    fn max_read_transfer_size(&self) -> Result<usize>;
```

Comment:
@cfrantz The need to advertise separate maximum length for read and write transactions made me think of another thing.  Maybe we need to also advertise the maximum length of operations for `run_eeprom_transaction()`, separate for the maximum from the existing `run_transaction()`.

For many implementations, the maximum data size for writes for `run_eeprom_transaction()` will 4-6 bytes less than the maximum write chunk size for the lower-level `run_transaction()`, as the opcode and address takes up a few bytes.  Dediprog, however, has a 16-byte limit on the low level function, but supports 64KB transactions through `run_eeprom_transaction()`.
I wonder if I should add two more methods, or maybe instead make one method that returns four values. (Or two methods that return two values each.)